src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/TableOfContentPageObject.java

```diff
@@ -54,16 +57,12 @@ public void clickOnTOCListElement(int index) {
     listOfLinks.get(index).click();
   }
 
-  public boolean isUserMovedToRightSection(int index) {
-    JavascriptExecutor js = (JavascriptExecutor) driver;
-    String h2PosString =
-        js.executeScript(
-            "return Math.floor($('section.article-body h2').eq(" + index + ").offset().top)")
-            .toString();
-    int h2Pos = Integer.parseInt(h2PosString);
-    int windowYPos =
-        Integer.parseInt(js.executeScript("return $(window).scrollTop()").toString());
-    return h2Pos == windowYPos;
+  public boolean isUserMovedToSectionByIndex(int sectionIndex) {
+    String sectionIndexString = String.valueOf(sectionIndex);
+    By sectionBy = By.cssSelector(section_selector.replace("__index__", sectionIndexString));
+    WebElement section = driver.findElement(sectionBy);
```

Comment:
This is string that represents a css selector, so this should be part of the UI map:
http://www.seleniumhq.org/docs/06_test_design_considerations.jsp#ui-mapping

A UI Map is a mechanism that stores all the locators for a test suite in one place for easy modification when identifiers or paths to UI elements change in the AUT. The test script then uses the UI Map for locating the elements to be tested. Basically, a UI map is a repository of test script objects that correspond to UI elements of the application being tested.

https://code.google.com/p/selenium/wiki/PageFactory